rasterio/_vsiopener.pyx

```diff
@@ -104,6 +102,9 @@ cdef void* pyopener_open(void *pUserData, const char *pszFilename, const char *p
     # ValueError if given one. We strip the mode in this case.
     except ValueError as err:
         file_obj = file_opener(filename, mode.rstrip("b"))
+    except Exception:
```

Comment:
We can be a little better about error handling here. If write mode is activated, it could fail because of a permissions problem. We can catch specifically `FileNotFoundError` and `PermissionError` and log the appropriate message. At the very least, we should log the exception that was caught.

Reply:
Yes... @groutr I'm still figuring out the Cython-Python-GDAL boundaries. The VSI plugin functions are called by GDAL and so the Python opener exceptions need to be turned into GDAL errors and put on GDAL's error stack.